crates/factor-key-value/src/lib.rs

```diff
@@ -131,6 +133,14 @@ impl AppState {
     }
 }
 
+#[async_trait]
```

Comment:
Can we document this trait? What guarantees are implementers expected to uphold?